From 66095ebbaf84b3edeb9644abf65e474815c50e2d Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Fri, 13 Jan 2012 12:33:07 +0530
Subject: [PATCH 092/609] arm:omap:am33xx : update TSC hwmod data

Original commit: 0e82a6b311ae038dc03f5291c93b42747762dbd7

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

This patch adds missing fields of hwmod data for TSC

Signed-off-by: Patil, Rachna <rachna@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c |   25 +++++++++++++++++++++++++
 1 files changed, 25 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 5201fce..375f289 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -67,6 +67,7 @@ static struct omap_hwmod am33xx_mmc2_hwmod;
 static struct omap_hwmod am33xx_spi0_hwmod;
 static struct omap_hwmod am33xx_spi1_hwmod;
 static struct omap_hwmod am33xx_elm_hwmod;
+static struct omap_hwmod am33xx_adc_tsc_hwmod;
 
 /*
  * Interconnects hwmod structures
@@ -227,6 +228,23 @@ static struct omap_hwmod_ocp_if am33xx_l4_wkup__gpio0 = {
 	.user		= OCP_USER_MPU | OCP_USER_SDMA,
 };
 
+/* L4 WKUP -> ADC_TSC */
+static struct omap_hwmod_addr_space am33xx_adc_tsc_addrs[] = {
+	{
+		.pa_start	= AM33XX_TSC_BASE,
+		.pa_end		= AM33XX_TSC_BASE + SZ_8K - 1,
+		.flags		= ADDR_TYPE_RT
+	},
+	{ }
+};
+
+static struct omap_hwmod_ocp_if am33xx_l4_wkup_adc_tsc = {
+	.master		= &am33xx_l4wkup_hwmod,
+	.slave		= &am33xx_adc_tsc_hwmod,
+	.addr		= am33xx_adc_tsc_addrs,
+	.user		= OCP_USER_MPU,
+};
+
 /* Master interfaces on the L4_WKUP interconnect */
 static struct omap_hwmod_ocp_if *am33xx_l4_wkup_masters[] = {
 	&am33xx_l4_wkup__gpio0,
@@ -258,6 +276,10 @@ static struct omap_hwmod_irq_info am33xx_adc_tsc_irqs[] = {
 	{ .irq = -1 }
 };
 
+static struct omap_hwmod_ocp_if *am33xx_adc_tsc_slaves[] = {
+	&am33xx_l4_wkup_adc_tsc,
+};
+
 static struct omap_hwmod am33xx_adc_tsc_hwmod = {
 	.name		= "adc_tsc",
 	.class		= &am33xx_adc_tsc_hwmod_class,
@@ -270,6 +292,9 @@ static struct omap_hwmod am33xx_adc_tsc_hwmod = {
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
+	.flags		= HWMOD_INIT_NO_RESET | HWMOD_INIT_NO_IDLE,
+	.slaves		= am33xx_adc_tsc_slaves,
+	.slaves_cnt	= ARRAY_SIZE(am33xx_adc_tsc_slaves),
 };
 
 /* 'aes' class */
-- 
1.7.5.4

